### PR TITLE
gnote: init at v44.0

### LINKED ISSUE #227679: Package request: Gnote
**Project description**
Gnote is a simple note-taking application for GNOME desktop environment. It allows you to capture your ideas, link them together using CamelCase-style links, group together in notebooks and some extra features for everyday use.

Notes can be printed or exported as HTML documents.

Gnote also supports synchronization, making it simple to use it on multiple devices.

Gnote started as a C++ port of Tomboy and has moved its own path later.

**Metadata**

* homepage URL: https://wiki.gnome.org/Apps/Gnote
* source URL: https://gitlab.gnome.org/GNOME/gnote
* license: gpl3+
* platforms: linux

### PR DESCRIPTION
**Pull request by jfvillablanca**
###### Description of changes

Based on the commits and reviews on #174827. It seems like there was no activity on the pr.

Resolves #227679

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/office/gnote/default.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  src = fetchurl {
+    url = "mirror://gnome/sources/gnote/${lib.versions.major version}/gnote-${version}.tar.xz";
+    hash = "sha256-bC50M87fwxhETsUZSi0zFrEIwZt8aUAMWeHi8Bj6z/g=";
+  };
+
+  buildInputs = [
+    glib
+  ];
+
+  nativeBuildInputs = [
```
Comment (Aleksanaa) [inline pkgs/applications/office/gnote/default.nix:37]:
```
  nativeBuildInputs = [
    desktop-file-utils
    docbook_xsl
    gettext
    itstool
    meson
    ninja
    pkg-config
    wrapGAppsHook
  ];

  buildInputs = [
    libxml2
    libxslt
    libuuid
    libsecret
    gtkmm3
    gspell
  ];
```

Comment (jfvillablanca) [reply pkgs/applications/office/gnote/default.nix:38]: Thank you

**Review by Aleksanaa — APPROVED** [review]
Works for me

**Comment by bobby285271** [comment]
Hmmm, I actually have some questions

- How is `docbook_xsl` used here? `grep -inr 'docbook'` does not return anything, I would expect docbook stuff like
  ```
  <?xml version="1.0"?>
  <!DOCTYPE book PUBLIC "-//OASIS//DTD DocBook XML V4.3//EN"
                        "http://www.oasis-open.org/docbook/xml/4.3/docbookx.dtd">
- Why 43.2 instead of 44.0? I know 43.2 is tagged later 44.0 but they are maintained on different branches and 44.0 should be the latest.

And a small nit, do you mind also adding an update script? We do mass GNOME updates in this way `nix-shell maintainers/scripts/update.nix --arg predicate '(path: pkg: pkg.updateScript.name or null == "gnome-update-script")' --argstr keep-going true --argstr commit true --argstr max-workers` so the package will be taken care.


```
  passthru = {
    updateScript = gnome.updateScript {
      packageName = pname;
    };
  };
```

**Comment by Aleksanaa** [comment]
Good question.

**Comment by bobby285271** [comment]
> move ... the `callPackage` to gnome's default.nix

No I think. We prefer new GNOME package [goes to top-level](https://github.com/NixOS/nixpkgs/pull/211138#discussion_r1071550374).

> move `gnote` to `pkgs/desktops/gnome/apps`

I don't aware of restrictions on this, the only difference should be `pkgs/desktops/gnome` has a CODEOWNER :see_no_evil:

**Comment by jfvillablanca** [comment]
Thank you for clarifying 😅

**Pull request merged**